File: arm.64/langs/csharp/src/boot.cs
namespace Pat;

public static class Booter
{
    public static List<Cabi> Boot(string? targetDir = null)
    {
        var engines = new List<Cabi>();
        string? envLib = Environment.GetEnvironmentVariable("PAT_LIB");
        if (!string.IsNullOrEmpty(envLib))
        {
            var c = Binder.Bind(envLib);
            if (c != null)
            {
                engines.Add(c);
                return engines;
            }
        }

        foreach (var file in Finder.Find(targetDir))
        {
            var c = Binder.Bind(file);
            if (c != null) engines.Add(c);
        }
        return engines;
    }
}

File: arm.64/langs/ocaml/src/boot.ml
let boot target_dir =
  match Sys.getenv_opt "PAT_LIB" with
  | Some lib when lib <> "" -> (
      match Bind.bind lib with
      | Some c -> [c]
      | None -> []
    )
  | _ ->
      Find.find target_dir
      |> List.filter_map Bind.bind
